src/main/java/io/anserini/search/SimpleSearcher.java

```diff
@@ -206,6 +206,17 @@ public void setBM25Similarity(float k1, float b) {
     searcher.setSimilarity(similarity);
   }
 
+  /**
+   * Returns the number of documents in the provided index file. Used to expose the total number of documents in Pyserini
+   *
+   * @return the number of documents in total
+   */
+
+   public Int getTotalNumDocuments(){
```

Comment:
primitive int instead if object Int.
I think there's already an `IndexSearcher` for you to read from? see line 96.

Reply:
I found line 96 was initialized to be a null and was assigned by some values in `setLMDirichletSimilarity`. Not sure if there are other assignments for this searcher

Reply:
IIRC, the code initializes an `IndexSearcher` on demand, as in: https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/SimpleSearcher.java#L308

You method should probably do the same also?

If there are any cases that would yield an NPE, can you please add a test case to catch, and then fix in code while you're at it?

Reply:
added test cases, still trying to catch possible NPE. For now, haven't encountered that.